diff --git a/grid/grid_classes.cc b/grid/grid_classes.cc
index 5e9930c..7e3f194 100644
--- a/grid/grid_classes.cc
+++ b/grid/grid_classes.cc
@@ -712,7 +712,11 @@ void GridInstance::on_sighup() {
 }
 
 void GridInstance::on_sigusr1() {
-    window.show(hint::Fullscreen);
+    if (window.get_visible()) {
+	window.hide();
+    } else {
+	window.show(hint::Fullscreen);
+    }
 }
 
 void GridInstance::on_sigint() {
